### ISSUE TITLE
`invalid-type-form` could hint at the appropriate type

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ty#593]
At least for the common case of `[int]`, we could suggest `list[int]`:

```
error[invalid-type-form]: List literals are not allowed in this context in a type expression
 --> autobot/schematics/use_generator/before.py:4:32
  |
4 | def compute_squares(n: int) -> [int]:
  |                                ^^^^^
5 |     squares: list[int] = []
6 |     for i in range(n):
  |
info: See the following page for a reference on valid type expressions:
info: https://typing.python.org/en/latest/spec/annotations.html#type-and-annotation-expressions
info: rule `invalid-type-form` is enabled by default
```

**Comment by j178** [comment]
Another common mistake like this happens with `tuple` in the return type:

```py
def f() -> (int, bool): pass
```

**Comment by benbaror** [comment]
I can try to do this as my 1st PR
